src/component/handlers/edit/editOnBeforeInput.js

```diff
@@ -167,17 +164,20 @@ function editOnBeforeInput(editor: DraftEditor, e: SyntheticInputEvent): void {
     newEditorState = EditorState.set(newEditorState, {
       nativelyRenderedContent: newEditorState.getCurrentContent(),
     });
-    // The native event is allowed to occur. To allow user onChange handlers to
-    // change the inserted text, we wait until the text is actually inserted
-    // before we actually update our state. That way when we rerender, the text
-    // we see in the DOM will already have been inserted properly.
-    editor._pendingStateFromBeforeInput = newEditorState;
-    setImmediate(() => {
-      if (editor._pendingStateFromBeforeInput !== undefined) {
-        editor.update(editor._pendingStateFromBeforeInput);
-        editor._pendingStateFromBeforeInput = undefined;
-      }
-    });
+
+    // Allow the native insertion to occur and update our internal state
+    // to match. If editor.update() does something like changing a typed
+    // 'x' to 'abc' in an onChange() handler, we don't want our editOnInput()
+    // logic to squash that change in favor of the typed 'x'. Set a flag to
+    // ignore the next editOnInput() event in favor of what's in our internal state.
+    editor._waitingOnInput = true;
+    editor.update(newEditorState, true);
+
+    if (isIE) {
+      setImmediate(() => {
+        editOnInput(editor);
```

Comment:
One of the major differences between my fix and flarnie's fix was that mine called `editOnInput` and hers didn't. Mine also caused https://github.com/textioHQ/editor/issues/582.

Have we validated that this doesn't regress 582? If it doesn't, why doesn't it?